esrally/utils/net.py

```diff
@@ -182,6 +182,13 @@ def download_http(url, local_path, expected_size_in_bytes=None, progress_indicat
         return expected_size_in_bytes
 
 
+def add_url_param(url, params):
+    url_parsed = urlparse(url)
+    query = dict(parse_qsl(url_parsed.query))
```

Comment:
Since we want to get a `dict` back and update it with `params`, should we use [parse_qs](https://docs.python.org/3.8/library/urllib.parse.html#urllib.parse.parse_qs) instead since it actually returns a `dict` rather an dictify the list that `parse_qsl` returns? According the above docs, if we do that, we should use `urlencode(query, doseq=True)` two lines below.